### PR TITLE
adding cookstyle for chef-recipes

### PR DESCRIPTION
**Pull request by pfuender**
Thanks for merging ;-)

Cheers,
Raphi

### CODE REVIEW

File: autoload/neomake/makers/ft/chef.vim [hunk 1]
```diff
@@ -4,11 +4,18 @@ endfunction
...
 function! neomake#makers#ft#chef#foodcritic() abort
     return {
       \ 'errorformat': '%WFC%n: %m: %f:%l',
       \ }
 endfunction
+
+function! neomake#makers#ft#chef#cookstyle() abort
+    return {
+      \ 'args': [ '-f', 'emacs', '--except', 'Style'],
+      \ 'errorformat': '%f:%l:%c: %t: %m,%E%f:%l: %m',
```
Comment (blueyed) [inline autoload/neomake/makers/ft/chef.vim]: The first one should also be prefixed with `%E` probably?!
Should everything be an error really?

File: autoload/neomake/makers/ft/chef.vim [hunk 2]
```diff
@@ -4,11 +4,18 @@ endfunction
 
 function! neomake#makers#ft#chef#EnabledMakers() abort
     let ruby_makers = neomake#makers#ft#ruby#EnabledMakers()
-    return ruby_makers + ['foodcritic']
+    return ruby_makers + ['foodcritic'] + ['cookstyle']
```
Comment (blueyed) [inline autoload/neomake/makers/ft/chef.vim]:
```vim
… + ['foodcritic', 'cookstyle']
```

File: autoload/neomake/makers/ft/chef.vim (hunk 1 above)
Comment (pfuender) [inline autoload/neomake/makers/ft/chef.vim]: Cookstyle is actually based on rubocop, so the errorformat is the same, so I quickly copy/pasted it from there. Having a second look at it, I can't really see any multiline errors at all (http://rubocop.readthedocs.io/en/latest/formatters/#emacs-style-formatter). I guess we can go with the following: 
`+      \ 'errorformat': '%f:%l:%c: %t: %m',`

**Comment by blueyed** [comment]
What about this then?
So you actually also use rubocops' postprocessor.

```diff
diff --git c/autoload/neomake/makers/ft/chef.vim i/autoload/neomake/makers/ft/chef.vim
index 13d83615..b6d506f7 100644
--- c/autoload/neomake/makers/ft/chef.vim
+++ i/autoload/neomake/makers/ft/chef.vim
@@ -4,7 +4,7 @@ endfunction
 
 function! neomake#makers#ft#chef#EnabledMakers() abort
     let ruby_makers = neomake#makers#ft#ruby#EnabledMakers()
-    return ruby_makers + ['foodcritic'] + ['cookstyle']
+    return ruby_makers + ['foodcritic', 'cookstyle']
 endfunction
 
 function! neomake#makers#ft#chef#foodcritic() abort
@@ -14,8 +14,7 @@ function! neomake#makers#ft#chef#foodcritic() abort
 endfunction
 
 function! neomake#makers#ft#chef#cookstyle() abort
-    return {
-      \ 'args': [ '-f', 'emacs', '--except', 'Style'],
-      \ 'errorformat': '%f:%l:%c: %t: %m,%E%f:%l: %m',
-      \ }
+    let maker = neomake#makers#ft#ruby#rubocop()
+    let maker.args += ['--except', 'Style']
+    return maker
 endfunction
```

(I did not want to push to your master, you should use a branch for PRs in the future)

File: autoload/neomake/makers/ft/chef.vim (hunk 1 above)
Comment (blueyed) [inline autoload/neomake/makers/ft/chef.vim]: What do the different types mean?  What is `C`?

**Comment by blueyed** [comment]
Why do you want to ignore `Style` by default?  It could just become something like `type=I` (info) instead?!

**Comment by pfuender** [comment]
The _Style_ is just very verbose and imho it's too much. I don't know that much about the other makers included, but do they also mention stuff like this?:
`
C: 2 trailing blank lines detected.
C: Use 2 (not 4) spaces for indentation.
C: Missing space after #.
C: Extra blank line detected.
`
Anyway, I've added them as type=I, as some of them are still useful. The errortypes are as per the rubocop source: https://github.com/bbatsov/rubocop/blob/022e7322b731b06bd31ce5ef1bac378b27551ed0/lib/rubocop/cop/severity.rb#L13.
I'm not sure about merging this with the rubocop maker (the errorformat part), I haven't seen any multiline errormessages, so I guess we can leave that aside or even change the rubocop-errorformat and merge these two. @blueyed, what do you think?
So all in all, I'd go for this version:

`
diff --git a/autoload/neomake/makers/ft/chef.vim b/autoload/neomake/makers/ft/chef.vim
index bbbb214..b0edc7e 100644
--- a/autoload/neomake/makers/ft/chef.vim
+++ b/autoload/neomake/makers/ft/chef.vim
@@ -4,7 +4,7 @@ endfunction

 function! neomake#makers#ft#chef#EnabledMakers() abort
     let ruby_makers = neomake#makers#ft#ruby#EnabledMakers()
-    return ruby_makers + ['foodcritic']
+    return ruby_makers + ['foodcritic', 'cookstyle']
 endfunction

 function! neomake#makers#ft#chef#foodcritic() abort
@@ -12,3 +12,11 @@ function! neomake#makers#ft#chef#foodcritic() abort
       \ 'errorformat': '%WFC%n: %m: %f:%l',
       \ }
 endfunction
+
+function! neomake#makers#ft#chef#cookstyle() abort
+    return {
+      \ 'args': [ '-f', 'emacs'],
+      \ 'errorformat': '%f:%l:%c: %t: %m',
+      \ 'postprocess': function('neomake#makers#ft#ruby#RubocopEntryProcess')
+      \ }
+endfunction
diff --git a/autoload/neomake/makers/ft/ruby.vim b/autoload/neomake/makers/ft/ruby.vim
index f585e6b..8b52d03 100644
--- a/autoload/neomake/makers/ft/ruby.vim
+++ b/autoload/neomake/makers/ft/ruby.vim
@@ -21,11 +21,20 @@ function! neomake#makers#ft#ruby#rubylint() abort
 endfunction

 function! neomake#makers#ft#ruby#RubocopEntryProcess(entry) abort
-    if a:entry.type ==# 'F'
-        let a:entry.type = 'E'
-    elseif a:entry.type !=# 'W' && a:entry.type !=# 'E'
-        let a:entry.type = 'W'
+    if a:entry.type ==# 'F'  " Fatal error which prevented further processing
+        let type = 'E'
+    elseif a:entry.type ==# 'E'  " Error for important programming issues
+        let type = 'E'
+    elseif a:entry.type ==# 'W'  " Warning for stylistic or minor programming issues
+        let type = 'W'
+    elseif a:entry.type ==# 'R'  " Refactor suggestion
+        let type = 'W'
+    elseif a:entry.type ==# 'C'  " Convention violation
+        let type = 'I'
+    else
+        let type = ''
     endif
+    let a:entry.type = type
 endfunction

 function! neomake#makers#ft#ruby#mri() abort
`

**Comment by blueyed** [comment]
LGTM basically.
However, the patch did not apply cleanly, so please just update the PR, and I will merge/amend it from there.

**Comment by pfuender** [comment]
Thanks for the comments, I pushed the newest version now.

**Comment by blueyed** [comment]
What about this?
Just let me know, and I can merge it already.

I think we should not change an unknown/unexpected type to be empty.

```
diff --git c/autoload/neomake/makers/ft/chef.vim i/autoload/neomake/makers/ft/chef.vim
index b0edc7ed..bd9079a0 100644
--- c/autoload/neomake/makers/ft/chef.vim
+++ i/autoload/neomake/makers/ft/chef.vim
@@ -15,8 +15,8 @@ endfunction
 
 function! neomake#makers#ft#chef#cookstyle() abort
     return {
-      \ 'args': [ '-f', 'emacs'],
+      \ 'args': ['-f', 'emacs'],
       \ 'errorformat': '%f:%l:%c: %t: %m',
-      \ 'postprocess': function('neomake#makers#ft#ruby#RubocopEntryProcess')
+      \ 'postprocess': function('neomake#makers#ft#ruby#RubocopEntryProcess'),
       \ }
 endfunction
diff --git c/autoload/neomake/makers/ft/ruby.vim i/autoload/neomake/makers/ft/ruby.vim
index 7bdb7948..2c41d549 100644
--- c/autoload/neomake/makers/ft/ruby.vim
+++ i/autoload/neomake/makers/ft/ruby.vim
@@ -14,19 +14,16 @@ endfunction
 
 function! neomake#makers#ft#ruby#RubocopEntryProcess(entry) abort
     if a:entry.type ==# 'F'  " Fatal error which prevented further processing
-        let type = 'E'
+        let a:entry.type = 'E'
     elseif a:entry.type ==# 'E'  " Error for important programming issues
-        let type = 'E'
+        let a:entry.type = 'E'
     elseif a:entry.type ==# 'W'  " Warning for stylistic or minor programming issues
-        let type = 'W'
+        let a:entry.type = 'W'
     elseif a:entry.type ==# 'R'  " Refactor suggestion
-        let type = 'W'
+        let a:entry.type = 'W'
     elseif a:entry.type ==# 'C'  " Convention violation
-        let type = 'I'
-    else
-        let type = ''
+        let a:entry.type = 'I'
     endif
-    let a:entry.type = type
 endfunction
 
 function! neomake#makers#ft#ruby#rubylint() abort
```

**Comment by pfuender** [comment]
That makes sense for the unknown types! So yes indeed, approved from my side :-)

**Comment by blueyed** [comment]
Thanks!
Please add it to the wiki.

**Pull request closed without merging**